### DISCUSSION TITLE
Using namespace config doesn't load the correct folder

### DISCUSSION THREAD

**Discussion by aprokopenko** [livewire/livewire#3042] [7. Bugs (archived)]
### Description
I have the application with 2 subsystems: the main app and the admin area.
Admin area is separated in different folder, so the structure looks like this:

```
/root
  |- app/                 # generic code like models and controllers for public/api part
  |- config/
  |- database/
  |- public/
  |- resources/
  |- storage/
  |- admin/
     |- app/               # php code, same structure as standard Laravel app with providers, Http, etc.
     |- config/           # admin specific config
     |- routes/           # admin specific routes
     |- resources/     # admin specific views and resources
```
Screenshot: https://i.imgur.com/YxNJFpR.png

Inside the composer, I have 2 namespaces App and Admin (the last one is linked to admin/app folder).
Inside AppServiceProvider I just register AdminServiceProvider,
AdminServiceProvider use standard Laravel features loadViewsFrom, loadTranslationsFrom, loadRoutesFrom with namespace `admin::`.

I set inside the config that I need components inside Admin\Http\Livewire. 

And I have such issues:
- After using the command to create component it was placed under App\Admin\Http\Livewire, which is wrong (I moved them manually where they should be)
- They are not loaded from my custom namespace, because Components resolver class has getNamespacePath static method, which has hardcoded app inside.
- Even LivewireComponentsFinder::getClassNames() also has hard-coded app/ path, which is wrong in my case.

I used to override LivewireComponentsFinder::getClassNames() and bind it manually to make it works.

### Exact steps to reproduce

- Create additional folder on the same level as app
- Register namespace to this folder
- Try to place components inside this folder

### Stripped-down, copy-pastable code snippets

Example of how I added support for admin (I know this is not very nice and pretty hard-coded version)^
```php
class LivewireComponentsFinder extends \Livewire\LivewireComponentsFinder
{
    /**
     * Get Class Names
     *
     * override vendor class to support custom location of admin folder.
     * currently vendor package searches only inside app/ folder and can't find outside.
     *
     * @return \Illuminate\Support\Collection
     */
    public function getClassNames()
    {
        return collect($this->files->allFiles($this->path))
            ->map(function (SplFileInfo $file) {
                if (! Str::startsWith($filepath = $file->getPathname(), $this->path)) {
                    return app()->getNamespace().
                        Str::of($filepath)
                            ->after(app_path().'/')
                            ->replace(['/', '.php'], ['\\', ''])->__toString();
                } else {
                    return 'Admin' . '\\' .
                        Str::of($filepath)
                            ->after($this->path . '/')
                            ->replace(['/', '.php'], ['\\', ''])->__toString();
                }
            })
            ->filter(function (string $class) {
                return is_a($class, Component::class, true)
                    && ! (new ReflectionClass($class))->isAbstract();
            });
    }
}
```

AdminServiceProvider:
```php
public function register()
{
    ...
    $this->app->singleton(\Livewire\LivewireComponentsFinder::class, function () use ($defaultManifestPath) {
            return new LivewireComponentsFinder(
                new Filesystem,
                config('livewire.manifest_path') ?: $defaultManifestPath,
                base_path('admin/app') // CUSTOM PATH HERE
            );
        });
}
```

**Context**
- Livewire version: 2.4.2
- Laravel version: 8.32.1
- Latest PHP FPM 7.4 buster docker image
- Browser: not related to browser

**Comment by francoism90** [comment]
@aprokopenko Did you ever solved this? I'm having the same question.

I want to use a custom path as well.

Thanks.

**Reply by aprokopenko** [reply]
Hi,

Actually, my solution is posted as a code snippet in the initial post.

In the next projects, we didn't use folder scan registration method and just registered components one by one in the service provider

**Reply by francoism90** [reply]
@aprokopenko Yep, that seems to be the best method. Thanks for your reply. :)
